rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -1124,6 +1124,41 @@ void save_saveNewFederationBtcUTXOs_utxosShouldBeSavedToStorage(FederationStorag
 
     }
 
+    @Test
+    void save_saveOldFederationBtcUTXOs_utxosShouldBeSavedToStorage() {
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        Address btcAddress = BitcoinTestUtils.createP2PKHAddress(networkParameters, "test");
+
+        DataWord oldFederationBtcUtxosKey = OLD_FEDERATION_BTC_UTXOS_KEY.getKey();
+
+        // Save utxos directly in storage.
+        List<UTXO> expectedUtxos = BitcoinTestUtils.createUTXOs(1, btcAddress);
+        storageAccessor.saveToRepository(oldFederationBtcUtxosKey, expectedUtxos, BridgeSerializationUtils::serializeUTXOList);
+
+        // Get utxos from method and check they are as expected
+        List<UTXO> actualUtxos = federationStorageProvider.getOldFederationBtcUTXOs();
+        assertEquals(1, actualUtxos.size());
+        assertEquals(expectedUtxos, actualUtxos);
+
+        List<UTXO> extraUtxos = BitcoinTestUtils.createUTXOs(1, btcAddress);
+
+        actualUtxos.addAll(extraUtxos);
+
+        federationStorageProvider.save(networkParameters, activations);
+
+        // Getting the utxos from storage to ensure they were stored in the storage.
+        List<UTXO> finalListOfUtxos = storageAccessor.getFromRepository(oldFederationBtcUtxosKey, BridgeSerializationUtils::deserializeUTXOList);
+
+        assertEquals(actualUtxos.size(), 2);
```

Comment:
Although it might seem strange now, in this line, the `actualUtxos.size()` part is actually the expected one. In the first assertion it was the `actual`, but in this second assertion it is the `expected`, although it's name doesn't reflect it.

Reply:
maybe add a comment saying that? is kind of confusing.
Also:
`assertEquals(actualUtxos.size(), finalListOfUtxos.size());`

Reply:
No need for that. If `assertEquals(actualUtxos, finalListOfUtxos);` is true, then their sizes are the same as well.
The reason why I'm doing things like `assertEquals(1, firstListOfUtxos.size());` and `assertEquals(2, finalListOfUtxos.size());` it's just to visually show that the list of uxtos has been updated to have more utxos, as we expect.

Reply:
i think you refactored it, looks good now!